### ISSUE TITLE
Side-panel / vertical tabs

### ISSUE THREAD

**Issue by rootex-** [warpdotdev/Warp#2155]
### Describe the solution you'd like?

```text
Side-panel tabs like in iterm2? 
Would be game-changer!
```

### How important is this feature to you?

5 (Can't work without it!)

**Comment by dannyneira** [comment]
Thanks for this feature request! Please +1 this request as it helps us gauge interest. We will post any updates here.

We currently have the ability to maximize an active pane into a tab, but it sounds like you want the opposite ( turn a Tab into a side-by-side Panes )

https://docs.warp.dev/features/windows/tabs
https://docs.warp.dev/features/windows/split-panes

**Comment by rootex-** [comment]
> Thanks for this feature request! Please +1 this request as it helps us gauge interest. We will post any updates here.
> 
> We currently have the ability to maximize an active Pane in a Tab, but it sounds like you want the opposite ( turn a Tab into a side-by-side/top-to-bottom Panes)
> 
> https://docs.warp.dev/features/windows/tabs https://docs.warp.dev/features/windows/split-panes

Thank you for the response.

I assume my explanation was too incomplete, sorry for that. 

What I actually offered was a vertical side-pannel (say, on the left) where all tab names (not tab contents) are vertically aligned in a column. Thus, it could have been a resizable column of tab names. 

So the user can vertically observe tab names from the top to bottom, resize the side panel to see longer tab names, switch between tabs vertically. 

Example: 

```
-----------------------------------------
Warp Window.                    |
-----------------------------------------|
tab1 |                                    |
---------            TAB1              |
tab2 |      CONTENTS         |
---------                                    |
tab3 |                                    |
------------------------------------------
```

**Comment by dannyneira** [comment]
I see thanks for clearing that up, so essentially vertical tabs like with Microsoft Edge browser for example

**Comment by rootex-** [comment]
Yes, exactly like Microsoft Edge browser's vertical tabs 👍🏽

**Comment by jandusek** [comment]
+1 this would indeed be a game-changer, the current horizontal tabs make it difficult to visually navigate anything more than 4 tabs

**Comment by galrosenthal** [comment]
+1 agreed, after working with iterm2 vertical tabs, and xTerm (on Windows) it feels like a must, im really not using warp as much as i want to because of this

**Comment by AlexR2D2** [comment]
+1  i am back to iTerm because if this feature

**Comment by evancauwenberg** [comment]
+1 after working with iterm2 vertical tabs, All my JetBrains IDE's also have this setup.
Still use iTerm more because of this but eager to change!

**Comment by apoorvsadana** [comment]
Any updates on this? Will really love to have this feature :)

**Comment by Mat-Longinow** [comment]
+1 to this, this would be huge

**Comment by amircohere** [comment]
Any updates on this feature? Thanks!!

**Comment by walterdis** [comment]
Any chance of implementation?

**Comment by shrikant14** [comment]
This would be a good feature to add

**Comment by ronkeiser** [comment]
I just want to pile on the pressure here. I love Warp very much, but this feature is crucial to my workflow.

**Comment by danielfstefan** [comment]
Same here, vertical tabs are really important for my workflow as well.

**Comment by AlexR2D2** [comment]
Tabs is about navigation. Navigation in app is the most important thing

**Comment by XuPengfei-1020** [comment]
@dannyneira Please, we need this feature!

**Comment by alxmrtn-gilead** [comment]
+1, hard to switch away from iTerm2 without this feature

**Comment by AlexR2D2** [comment]
> +1, hard to switch away from iTerm2 without this feature

looks like they just don't want to lure the iTerm users ;)

**Comment by dabobert** [comment]
i want to use warp but i really need this

**Comment by evancauwenberg** [comment]
+1 .. still on iTerm for this, eager to change ;)

**Comment by kamil-malinski** [comment]
please consider this, a lot of people would love it (me including)

**Comment by joakimkingstrom** [comment]
Please, please, please add this feature!

**Comment by jwaldrip** [comment]
SAME, please add!

**Comment by NipunManocha-JISR** [comment]
<img width="1485" alt="Image" src="https://github.com/user-attachments/assets/3a40c45f-9705-4806-a0d9-32d930b4983e" />

Even the AI assumes this feature is already present.

**Comment by tripol** [comment]
> Thanks for this feature request! Please add a :+1: to the original post at the top to signal that you want this feature, and then hit the subscribe button if you'd like to be notified.
> 
> We currently have the ability to maximize an active Pane in a Tab, but it sounds like you want the opposite ( turn a Tab into a side-by-side/top-to-bottom Panes)
> 
> https://docs.warp.dev/features/windows/tabs https://docs.warp.dev/features/windows/split-panes

My suggestion would be more in the direction of the vertical tabs functioning as 'Workspaces', similar to how we Arc/Zen's browser's workspace feature. So that

1) Vertical tabs: For workspaces, reducing the need to maintain multiple windows. Each vertical tab/workspace can then maintain their respective horizontal tabs as needed by the user
2) Horizontal tabs: Normal tabs within each workspace

**Comment by jordangullen** [comment]
+1 - loving Warp but it really needs vertical tabs.
@dannyneira - any updates on this feature being added to the roadmap?

**Comment by egold** [comment]
Since almost all screens are widescreen now, with a greater aspect ratio differential than the only 4:3 ratio, vertical pixels are "worth" 1.6X (16:10) / 1.78X (16:9) of a horizontal pixel. Arc browser for example only has side tabs for this reason. It's a great evidence-driven approach to tab display.

**Comment by jere-co** [comment]
This would drastically improve my workflow as well! It's mostly an issue when working with several projects at once and keeping tabs open for all. Currently it becomes impossible to see the tab name with enough tabs open.

**Comment by fiveonGit** [comment]
Wow, 3 years since this was opened and still no implementation from the team. 🫤
Please, is there any update on this?

**Comment by rajpootathar** [comment]
bump, i need this i am running 5 agentic tabs.

**Comment by SENTINELITE** [comment]
Just chiming in here as well. I often have a lot of windows open, each with their own tabs. Would be great to have a "Spaces" or "Projects" vertical sidebar, ideally & mainly so it can reduce RAM/CPU usage on boot, etc.

**Comment by aarush-kukreja** [comment]
+1

**Comment by josepmesteves** [comment]
+1 please, add this. I can't manage my tabs with ease using the horizontal bar at the top. It has always been a pain to have more than 5 tabs in warp.


it's a pain big enough that i would implement this myself if i had access to the codebase

**Comment by Molunerfinn** [comment]
Hi @dannyneira , could we know the expected release date for this feature?

**Comment by dannyneira** [comment]
Vertical tabs are live and available on Warp Preview! https://www.warp.dev/download-preview

Please try it and let us know your feedback in our community Slack. https://go.warp.dev/join-preview

**Comment by daviziks** [comment]
I'm using it for 2 days, no issues at all, pretty good!
Arch Linux + GNOME Wayland

**Comment by AlexR2D2** [comment]
Oh, event a terminal app asks me to login, no, no way, enough!

**Comment by zachbai** [comment]
Vertical tabs are officially available in the latest stable release 🎊

**Comment by egold** [comment]
OMG this is so much better. I went from unreadable tab titles to glorious readable comfortable tabs with plenty of breathing room. 

This makes so much more sense in the 16:9 and wider monitor days. Which, to be fair, have been here for well over 20 years. Kudos!

**Comment by ragamufin** [comment]
A big THANK YOU to the team!
Going to be using it for the next couple days instead of cmux
Was hoping for both vertical and horizontal tabs but will see how this goes
Been on cmux for the past month or so and while it has been tremendous to have everything grouped nicely and just "know" where things are, I miss the aesthetic and million little touches of warp, the "feel" of it

Will report back how it goes
big thanks again

**Comment by ahuth** [comment]
Agreed that vertical tabs, with each still having their own set of horizontal tabs, would be better.

But at least now we can get most of the way their with splitting the pane for each vertical tab.
